internal/testrunner/runners/system/runner.go

```diff
@@ -1005,6 +1091,94 @@ func (r *runner) prepareScenario(ctx context.Context, config *testConfig, servic
 	return &scenario, nil
 }
 
+func (r *runner) setupService(ctx context.Context, config *testConfig, serviceOptions servicedeployer.FactoryOptions, svcInfo servicedeployer.ServiceInfo, agentInfo agentdeployer.AgentInfo, agentDeployed agentdeployer.DeployedAgent, state ServiceState) (servicedeployer.DeployedService, servicedeployer.ServiceInfo, error) {
+	logger.Debug("setting up service...")
+	if r.options.RunTearDown || r.options.RunTestsOnly {
+		svcInfo.Test.RunID = state.ServiceRunID
+		svcInfo.OutputDir = state.ServiceOutputDir
+	}
+
+	// By default using agent running in the Elastic stack
+	svcInfo.AgentNetworkName = stack.Network(r.options.Profile)
+	if agentDeployed != nil {
+		svcInfo.AgentNetworkName = agentInfo.NetworkName
+	}
+
+	// Set the right folder for logs execpt for custom agents that are still deployed using "servicedeployer"
+	if r.options.RunIndependentElasticAgent && agentDeployed != nil {
+		svcInfo.Logs.Folder.Local = agentInfo.Logs.Folder.Local
+	}
+
+	// In case of custom agent (servicedeployer) enabling independent agents, update serviceOptions to include test policy too
+	if r.options.RunIndependentElasticAgent {
+		serviceOptions.PolicyName = agentInfo.Policy.Name
+	}
```

Comment:
Currently, it is not set the Agent Policy name in all the scenarios, to avoid changing the current behaviour. Currently, in servicedeployer it is used the default policy `Elastic-Agent (elastic-package)`